internal/cache/inmemory.go

```diff
@@ -1,48 +1,57 @@
 package cache
 
 import (
-	"errors"
-	"fmt"
 	"time"
 
-	"github.com/ReneKroon/ttlcache/v2"
+	"github.com/jellydator/ttlcache/v3"
 	"github.com/open-policy-agent/opa/server/types"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
-var errNotSupportedType = errors.New("value type not supported")
-
 type inmemory struct {
-	tc ttlcache.SimpleCache
+	tc *ttlcache.Cache[string, []byte]
 }
 
-func NewInMemoryCache(expire int32) Cacher {
-	tc := ttlcache.NewCache()
-	_ = tc.SetTTL(time.Duration(expire * int32(time.Second)))
+func NewInMemoryCache(expire int32) CacherWithMetrics {
+	expireDuration := time.Duration(int64(expire) * int64(time.Second))
```

Comment:
To clarify this one more time: Didn't we agree on turning `expire` to `int64` to overcome the cast situation?

Reply:
The `expire` parameter to the function and the `Memcached.Expire` attribute in the configuration can stay `int32`, because they just need to be able to express "seconds". It only needs to be 64bit wide once it gets multiplied with the nanosecond-precision `time.Duration` values.